Source/SharpNeedle/HedgehogEngine/Needle/TextureStreaming/Package.cs

```diff
@@ -0,0 +1,200 @@
+﻿using SharpNeedle.Utilities;
+using System.IO;
+
+namespace SharpNeedle.HedgehogEngine.Needle.TextureStreaming;
+
+public class Package : ResourceBase, IBinarySerializable, IStreamable
+{
+    public struct EntryInfo : IBinarySerializable
+    {
+        public int Hash { get; set; }
+        public int BlockIndex { get; set; }
+        public int MipLevels { get; set; }
+        public ushort Width { get; set; }
+        public ushort Height { get; set; }
+        public string Name { get; set; }
```

Comment:
Name should be a property with a backing field where the setter should update the hash property.

Reply:
Adjusted